### PR TITLE
PCHR-2543: Fix style issues generated by Bootstrap migration

### PR DESCRIPTION
**Pull request by reneolivo**
## Overview
This PR fixes several issues that emerged after migrating some of the admin screens to Bootstrap. 

This PR depends on https://github.com/compucorp/org.civicrm.shoreditch/pull/38

#### Note
Please ignore black * on the "After" screens.  That issue is fixed here https://github.com/civicrm/civihr/pull/2141/commits/00872e311e769bcdd48020c66c1a672b4a057210 and here https://github.com/compucorp/org.civicrm.shoreditch/pull/38/commits/56739ebff76f2dcbab9d53001cbcfaba078e974b .

#### Before
download 26

#### After
download 17

### Padding top of form panel changed to 20px.
Also the margin bottom of the titles was changed to 20px as well.

#### Before
download 27

#### After
download 18

### Submit buttons were moved to panel's footer and the right spacing was reduced to 20px.
Some of the buttons were inside panel's bodies.

#### Before
download 29

#### After
download 20

### Date pickers were breaking.
This is was introduced by not running backstop at the appropriate branches. To fix this issue `crm-form-block` was added to all forms instead of extending `.crm-form-text` with `.form-control`

#### Before
download 30

#### After
download 21

#### Before
download 33

#### After
download 22

## Technical Details
* Added `leaveandabsence.css` file to several list pages so they could override buttons margins and follow the 20px rule:
`CRM_Core_Resources::singleton()->addStyleFile('uk.co.compucorp.civicrm.hrleaveandabsences', 'css/leaveandabsence.css');`
* Form panels now have 20px padding on top and button, the first h3 has no margin top, other h3s have a margin bottom of 20px, and footer buttons align to the right without any margin so they have an exact margin of 20px in relation to the panel:
```sass
.crm-leave-and-absences-form-block {
  .panel-body {
    // Panel has 20px on top and bottom:
    padding-top: $crm-panel-nested-indent;
    padding-bottom: $crm-panel-nested-indent;

    // No margin on first h3:
    .row:first-child h3 {
      margin-top: 0;
    }

   // Other h3s have 20px margin bottom:
    h3 {
      margin-bottom: $crm-panel-nested-indent;
    }
  }

  .panel-footer .crm-button:last-child {
    // Remove margin-right from last button so it aligns to panel:
    margin-right: 0;
  }
}
```

* On L&A import screen the date formats started to look bold (because they are inside a label) so this css rule reverts them back to look normal:
```scss
table.date-formats {
    label {
      font-weight: normal !important;
    }
}
```

* For the general settings issue, the label description was using the class `crm-marker` which is used to make * red and bold. The class was changed to `crm-marker-description` (which explains why the input is required):
```php
'label' =>  'Relationship types that are allowed to approve leave: <br/>
                   <span class="crm-marker-description">(Please note that CiviHR admin'.........</span>
```

* Form titles were turned into bootstrap's panel headings ([see diff](https://github.com/civicrm/civihr/pull/2141/files#diff-8d409d228a17906ead9b8cfb82fe3e98R2))

The following Backstop JS scenarios were run successfully from this branch against la-milestone-3:

* leave-absence-admin-absence-periods.json
* leave-absence-admin-absence-types.json
* leave-absence-admin-absence-types.json
* leave-absence-admin-general-settings.json
* leave-absence-admin-public-holidays.json

## Comments

The entitlement calculation filters were broken because some classes were removed. These classes were added back and now the filters work as they should.

### Before
peek 2017-09-14 12-36

### After
anim

### BackstopJS scenario rename

Backstop **public-holidays.json** scenario was renamed to **leave-absence-admin-public-holidays.json**. The reason is so it follows naming convention.

---

- [x] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/templates/CRM/HRLeaveAndAbsences/Form/ManageEntitlements.tpl [hunk 1]
```diff
@@ -16,15 +16,15 @@
         <div class="col-sm-4 col-sm-offset-4">
           <div class="btn-group btn-group-sm" data-toggle="buttons">
             <label class="btn btn-default">
-              <input type="radio" name="override-filter" value="1">
+              <input type="radio" class="override-filter" name="override-filter" value="1">
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/templates/CRM/HRLeaveAndAbsences/Form/ManageEntitlements.tpl]: Removing `name` doesn't break the behaviour?

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/pages/_admin-leave-and-absences-list-base.scss [hunk 2]
```diff
@@ -0,0 +1,5 @@
+&.crm-leave-and-absences-list-block {
+  .panel-body .button.pull-right {
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/scss/pages/_admin-leave-and-absences-list-base.scss]: Empty line is missing.

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/pages/_admin-leave-and-absences-form-base.scss [hunk 3]
```diff
@@ -1,8 +1,18 @@
-#crm-container .crm-leave-and-absences-form-block {
-  .crm-submit-buttons {
-    .crm-button-type-next,
-    .crm-button-type-cancel {
-      float: right !important;
+.crm-leave-and-absences-form-block {
+  .panel-body {
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/scss/pages/_admin-leave-and-absences-form-base.scss]: Empty line is missing

File: uk.co.compucorp.civicrm.hrleaveandabsences/templates/CRM/HRLeaveAndAbsences/Form/ManageEntitlements.tpl (hunk 1 above)
Comment (reneolivo) [reply uk.co.compucorp.civicrm.hrleaveandabsences/templates/CRM/HRLeaveAndAbsences/Form/ManageEntitlements.tpl]: I made a bad migration. Will check if the name is required (it was present waaaay before), but the class should go on the button group, not at the input level.

Reference: https://github.com/civicrm/civihr/blob/dc8e7508c27ca58ac95e1f2b2b23ed055a7735a7/uk.co.compucorp.civicrm.hrleaveandabsences/templates/CRM/HRLeaveAndAbsences/Form/ManageEntitlements.tpl#L18

Comment (reneolivo) [reply uk.co.compucorp.civicrm.hrleaveandabsences/templates/CRM/HRLeaveAndAbsences/Form/ManageEntitlements.tpl]: ok, sorry, override-filter is ok. The name is not necessary (checked PHP files, none made reference to this field). Will remove name attribute.

**Pull request merged**